pkg/solver/matcher.go

```diff
@@ -18,58 +19,311 @@ func (a ListOfResourceOffers) Less(i, j int) bool {
 }
 func (a ListOfResourceOffers) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
 
+type AllowlistItem struct {
+	Enabled  bool
+	ModuleID string
+	Version  string
+}
+
+func loadAllowlist(filepath string) ([]AllowlistItem, error) {
+		file, err := os.Open(filepath)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
+	
+		bytes, err := ioutil.ReadAll(file)
+		if err != nil {
+			return nil, err
+		}
+	
+		var allowlist []AllowlistItem
+		if err := json.Unmarshal(bytes, &allowlist); err != nil {
+			return nil, err
+		}
+	
+		return allowlist, nil
+	}
+
+	// strip out https- keep the org and module in place
+	// https://lilypad.tech/cowsay -> lilypad.tech/cowsay
+	// Assume module ID has the shape of the org/repo(eg lilypad-tech/cowsay) if string
+	func moduleMatch(moduleID, allowlistModuleID string) bool {
+		return moduleID == allowlistModuleID
+	}
+	
+	func versionMatch(version, allowlistVersion string) bool {
+		if strings.Contains(allowlistVersion, "*") {
+			prefix := strings.TrimSuffix(allowlistVersion, "*")
+			return strings.HasPrefix(version, prefix)
+		}
+		return version == allowlistVersion
+	}
+
+	
 // the most basic of matchers
 // basically just check if the resource offer >= job offer cpu, gpu & ram
-// if the job offer is zero then it will match any resource offer
+
+/*// if the job offer is zero then it will match any resource offer
 func doOffersMatch(
 	resourceOffer data.ResourceOffer,
 	jobOffer data.JobOffer,
-) bool {
-	if resourceOffer.Spec.CPU < jobOffer.Spec.CPU {
+) bool */
+func doOffersMatch(resourceOffer data.ResourceOffer, jobOffer data.JobOffer, allowlist []AllowlistItem, allowlistEnabled bool) bool {
+	if !allowlistEnabled {
```

Comment:
Does this mean: "if the allow list feature is not enabled" then all offers match?